File: common/com.raytheon.uf.common.dataplugin.events.hazards/utility/common_static/base/HazardServices/python/events/GenericSessionObjectManager.py
# #
# This software was developed and / or modified by the
# National Oceanic and Atmospheric Administration (NOAA), 
# Earth System Research Laboratory (ESRL), 
# Global Systems Division (GSD), 
# Evaluation & Decision Support Branch (EDS)
# 
# Address: Department of Commerce Boulder Labs, 325 Broadway, Boulder, CO 80305
# #

#
# Manager for generic session objects, allowing them to be stored and
# retrieved. Unlike generic registry objects, these objects are not
# persisted between sessions, nor are they shared between clients.
#    
#     SOFTWARE HISTORY
#    
#    Date            Ticket#       Engineer       Description
#    ------------    ----------    -----------    --------------------------
#    May 30, 2018    14791         Chris.Golden   Initial creation.
#
import JUtil

class GenericSessionObjectManager(object):
    
    def __init__(self, javaObject):
        '''
        @summary Construct an instance wrapping a Java GenericSessionObjectManager
        object.
        @param javaObject Java GenericSessionObjectManager instance to be wrapped.
        '''
        self._javaObject = javaObject

    def get(self, key):
        '''
        @summary Get the generic session object associated with the specified key.
        @param key Key for which to fetch the value.
        @return Value associated with the key, or None if no such value is found.
        '''
        value = self._javaObject.get(key)
        return JUtil.javaObjToPyVal(value) if value is not None else None

    def set(self, key, value):
        '''
        @summary Create an entry associating the specified key with the specified
        value.
        @param key Key for which to create the new entry.
        @param value Value to be associated with the key.
        '''
        value = JUtil.pyValToJavaObj(value) if value is not None else None
        return self._javaObject.set(key, value)

    def getWrappedJavaObject(self):
        '''
        @summary Get the Java object, which will be a GenericSessionObjectManager,
        that is wrapped by this instance.
        @return Java object wrapped by this instance.
        '''
        return self._javaObject

File: common/com.raytheon.uf.common.dataplugin.events/src/com/raytheon/uf/common/dataplugin/events/GenericSessionObjectManager.java
/**
 * This software was developed and / or modified by the
 * National Oceanic and Atmospheric Administration (NOAA), 
 * Earth System Research Laboratory (ESRL), 
 * Global Systems Division (GSD), 
 * Evaluation & Decision Support Branch (EDS)
 * 
 * Address: Department of Commerce Boulder Labs, 325 Broadway, Boulder, CO 80305
 */
package com.raytheon.uf.common.dataplugin.events;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

/**
 * Generic session object manager, used to hold generic objects for the duration
 * of a session. These are not shared between clients, nor do they persist
 * between sessions.
 * <p>
 * Implementation note: This class does not subclass one of the {@link Map}
 * implementations, nor may one of those implementations be used in its place,
 * due to the fact that it needs to be converted to a Python
 * <code>GenericSessionObjectManager</code>, and not some sort of Python
 * dictionary, when it crosses the Java/Python boundary and is used in a
 * recommender (for example). If it were a Java <code>Map</code>, it would be
 * automatically converted to a Python dictionary, which could incur a
 * significant performance cost if it held large and complex Java objects. By
 * using an object that is not an instance of any <code>Map</code>
 * implementation, this may be avoided.
 * </p>
 * 
 * <pre>
 *
 * SOFTWARE HISTORY
 *
 * Date         Ticket#    Engineer      Description
 * ------------ ---------- ------------- --------------------------
 * May 30, 2018   14791    Chris.Golden  Initial creation.
 *
 * </pre>
 *
 * @author Chris.Golden
 */
public class GenericSessionObjectManager implements Serializable {

    // Private Variables

    /**
     * Backing map.
     */
    private final Map<String, Serializable> map = new HashMap<>();

    // Public Methods

    /**
     * Get the session object associated with the specified key.
     * 
     * @param key
     *            Key for which to fetch a session object.
     * @return Session object associated with the key, or <code>null</code> if
     *         no associated object is found.
     */
    public Serializable get(String key) {
        return map.get(key);
    }

    /**
     * Set the session object associated with the specified key.
     * 
     * @param key
     *            Key for the new object.
     * @param value
     *            Object to be associated with the key.
     */
    public void set(String key, Serializable value) {
        map.put(key, value);
    }
}
